Library/Homebrew/formula.rb

```diff
@@ -598,7 +598,16 @@ def rack
   # All currently installed prefix directories.
   # @private
   def installed_prefixes
-    rack.directory? ? rack.subdirs.sort : []
+    prefixes = rack.directory? ? rack.subdirs : []
+
+    prefixes += (aliases + Array(oldname)).map do |alias_name|
+      rack_alias = HOMEBREW_CELLAR/alias_name
+      next unless rack_alias.directory?
+
+      rack_alias.subdirs
+    end.compact
+
+    prefixes.flatten.sort_by(&:basename)
```

Comment:
```suggestion
    prefixes += (aliases + Array(oldname)).flat_map do |alias_name|
      rack_alias = HOMEBREW_CELLAR/alias_name
      next unless rack_alias.directory?

      rack_alias.subdirs
    end.compact

    prefixes.sort_by(&:basename)
```

Sorry to bike shed. Feel free to pass on this if desired.

Reply:
Forgot `flat_map` even existed.